File: headapps/app/src/lib/factories/ford-explorer-factory.ts
import { VehicleBuilder } from "../builders/vehicle-builder";
import { CarFactory } from "./car-factory";

export class FordExplorerFactory extends CarFactory {
  protected createBuilder(): VehicleBuilder {
    return new VehicleBuilder()
      .setColor("Red")
      .setBrand("Ford")
      .setModel("Explorer");
  }
}

File: DesignPatterns/FactoryMethods/FordExplorerFactory.cs
using DesignPatterns.ModelBuilders;

namespace DesignPatterns.FactoryMethods
{
    public class FordExplorerFactory : CarFactory
    {
        public override string ModelKey => "Explorer";

        protected override CarBuilder CreateBuilder()
        {
            return new CarBuilder()
                .setColor("Red")
                .setBrand("Ford")
                .setModel("Explorer");
        }
    }
}
